File: Python/src/my_array.py
class MyArray:
    def __init__(self, capacity):
        # Initialize an array with given capacity
        self.capacity = capacity
        self.arr = []

    def add(self, value):
        # Adds a value to the array if there is space
        if len(self.arr) >= self.capacity:
            raise IndexError("Array is full!")
        self.arr.append(value)

    def get(self, index):
        # Gets the element at the given index
        if index >= len(self.arr):
            raise IndexError("Index out of bounds")
        return self.arr[index]

    def size(self):
        # Returns the size of the array
        return len(self.arr)

File: CSharp/src/MyArray.cs
namespace CSharp;

public class MyArray
{
    private int[] _array;
    private int _size;
    private int _capacity;

    public MyArray(int capacity = 4)
    {
        _capacity = capacity;
        _array = new int[capacity];
        _size = 0;
    }

    // Add an element to the array
    public void Add(int value)
    {
        if (_size == _capacity)
        {
            Resize();
        }
        _array[_size] = value;
        _size++;
    }

    // Resize the array when it is full
    private void Resize()
    {
        _capacity *= 2;
        int[] newArray = new int[_capacity];
        for (int i = 0; i < _size; i++)
        {
            newArray[i] = _array[i];
        }
        _array = newArray;
    }

    // Get an element by index
    public int Get(int index)
    {
        if (index < 0 || index >= _size)
        {
            throw new IndexOutOfRangeException("Index is out of bounds");
        }
        return _array[index];
    }

    // Get the current size of the array
    public int Size()
    {
        return _size;
    }
}

